### PR TITLE
merge(pbts): merge branch `main` into `feature/pbts`

### PR DESCRIPTION
**Pull request by sergio-mena**
Part of #1731

N.B.: This is a merge commit bringing all recent changes on `main` into feature branch `feature/pbts`, so no detailed review is needed, as the changes have already been reviewed on `main`.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: internal/consensus/common_test.go [hunk 1]
```diff
@@ -523,12 +527,6 @@ func randStateWithAppWithHeight(
 	return randStateWithAppImpl(nValidators, app, c)
 }
 
-func randStateWithAppWithBFTTime(nValidators int) (*State, []*validatorStub) {
-	c := test.ConsensusParams()
-	c.Feature.PbtsEnableHeight = 0 // Disable PBTS
-	return randStateWithAppImpl(nValidators, kvstore.NewInMemoryApplication(), c)
-}
-
```
Comment (lasarojc) [inline internal/consensus/common_test.go:531]: Is this supposed to be removed?

Comment (sergio-mena) [reply internal/consensus/common_test.go:531]: The linter told me it's unused 🤷‍♂️

File: internal/consensus/pbts_test.go [hunk 2]
```diff
@@ -529,12 +529,6 @@ func TestPBTSTooFarInTheFutureProposal(t *testing.T) {
 	require.Nil(t, results.height2.prevote.BlockID.Hash)
 }
 
-func pbtsFromHeightParams(height int64) types.FeatureParams {
-	p := types.DefaultFeatureParams()
-	p.PbtsEnableHeight = height
-	return p
-}
-
```
Comment (lasarojc) [inline internal/consensus/pbts_test.go:537]: ditto

Comment (sergio-mena) [reply internal/consensus/pbts_test.go:537]: Yeah, same, linter ranted, it's unused

File: internal/consensus/pbts_test.go [hunk 3]
```diff
@@ -612,11 +607,12 @@ func TestPBTSEnableHeight(t *testing.T) {
 				proposal.Timestamp = time.Now()
 			}
 			signProposal(t, proposal, chainID, vss[proposer])
-			cs.SetProposalAndBlock(proposal, block, blockParts, "p")
+			err := cs.SetProposalAndBlock(proposal, block, blockParts, "p")
+			require.NoError(t, err)
 			ensureProposal(proposalCh, height, round, blockID)
 		}
 
-		delta := time.Now().Sub(block.Time)
```
Comment (lasarojc) [inline internal/consensus/pbts_test.go:619]: this should not be reverted

Comment (sergio-mena) [reply internal/consensus/pbts_test.go:619]: It's before my PR, so we're not "needing" `cmttime.Now` yet. I'll be fixing this in my PR, when I rebase it on top of this

**Review by lasarojc — APPROVED** [review]

**Comment by sergio-mena** [comment]
Thanks @lasarojc for this thorough review!

File: internal/consensus/common_test.go (hunk 1 above)
Comment (cason) [reply internal/consensus/common_test.go:531]: I removed the use of it in #2492

**Pull request merged**